# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: e8fac91d4012
# Full Hash: e8fac91d4012b0ac5acce2d73847335d3828321b
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2021-01-04 09:08:57
# Regressor Bug: 1681043
# File Overlap Count: 2
# Description:
#   Bug 1681043 - P1. Launch RDD process early. r=alwu
#   
#   We can launch the RDD process early now that it is a fully asynchronous process and doesn't block anything.
#   
#   This will allow to retrieve the decoding capabilities of the RDD process right away.
# ==============================================================================

diff -r 5d453db46486 -r e8fac91d4012 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Sun Jan 03 20:32:29 2021 +0000
+++ b/dom/ipc/ContentParent.cpp	Sun Jan 03 23:20:38 2021 +0000
@@ -66,6 +66,7 @@
 #include "mozilla/PresShell.h"
 #include "mozilla/ProcessHangMonitor.h"
 #include "mozilla/ProcessHangMonitorIPC.h"
+#include "mozilla/RDDProcessManager.h"
 #include "mozilla/ScopeExit.h"
 #include "mozilla/ScriptPreloader.h"
 #include "mozilla/Services.h"
@@ -2889,6 +2890,12 @@
 
   gpm->AddListener(this);
 
+  if (StaticPrefs::media_rdd_process_enabled()) {
+    // Ensure the RDD process has been started.
+    RDDProcessManager* rdd = RDDProcessManager::Get();
+    rdd->LaunchRDDProcess();
+  }
+
   nsStyleSheetService* sheetService = nsStyleSheetService::GetInstance();
   if (sheetService) {
     // This looks like a lot of work, but in a normal browser session we just